e2e2/internal/framework_extensions/conditions.go

```diff
@@ -41,3 +42,17 @@ func (c *ConditionExtension) DaemonSetReady(daemonset k8s.Object) apimachinerywa
 		return
 	}
 }
+
+func (c *ConditionExtension) JobSucceeded(job k8s.Object) apimachinerywait.ConditionWithContextFunc {
+	return func(ctx context.Context) (done bool, err error) {
+		if err := c.resources.Get(ctx, job.GetName(), job.GetNamespace(), job); err != nil {
+			return false, err
+		}
+		status := job.(*batchv1.Job).Status
+		spec := job.(*batchv1.Job).Spec
+		if status.Succeeded == *spec.Completions {
+			done = true
```

Comment:
nit: I would rather negate condition and return false here than use named return vars

Reply:
Did it in the latest revision.